mkosi/config.py

```diff
@@ -905,23 +905,22 @@ def config_default_repository_key_fetch(namespace: argparse.Namespace) -> bool:
     if detect_distribution()[0] != Distribution.ubuntu:
         return False
 
+    def needs_repository_key_fetch(distribution: Distribution) -> bool:
+        return distribution == Distribution.arch or distribution.is_rpm_distribution()
```

Comment:
Let's move this to the beginning of the function, having function definitions in the middle always strikes me a bit unexpected.